pkg/client/introduce/example_test.go

```diff
@@ -0,0 +1,306 @@
+						fmt.Println("svc.HandleInbound", err)
+						return
+					}
+				}
+
+				// mocks didexchange logic by calling directly InvitationReceived function from the introduce service
+				if err := svc.InvitationReceived(stateMsg(inboundMsg.msg)); err != nil {
+					fmt.Println("InvitationReceived", err)
+					return
+				}
+			}
+		}
+	}()
+
+	clientStorageProvider := storageMocks.NewMockProvider(ctrl)
+	clientStorageProvider.EXPECT().
+		OpenStore(introduce.Introduce).
+		Return(fakeStore(ctrl), nil).
+		AnyTimes()
+
+	introduceClientProvider := clientIntroduceMocks.NewMockProvider(ctrl)
+	introduceClientProvider.EXPECT().
+		Service(introduce.Introduce).
+		Return(svc, nil)
+	introduceClientProvider.EXPECT().
+		StorageProvider().
+		Return(clientStorageProvider)
+
+	return introduceClientProvider, done
+}
+
+// nolint: gocyclo
+func Example() {
+	wg := sync.WaitGroup{}
+	defer wg.Wait()
+
+	var transport = map[string]chan transportMsg{
+		Alice: make(chan transportMsg),
+		Bob:   make(chan transportMsg),
+	}
+
+	// prepare the clients contexts
+	aliceCtx, aliceDone := provider(Alice, transport)
+	bobCtx, bobDone := provider(Bob, transport)
+
+	// create client
+	alice, err := New(aliceCtx, nil)
+	if err != nil {
+		fmt.Println("Alice client:", err)
+	}
+
+	// register for action events
+	aliceActions := make(chan service.DIDCommAction)
+
+	err = alice.RegisterActionEvent(aliceActions)
+	if err != nil {
+		fmt.Println("Alice RegisterActionEvent:", err)
+	}
+
+	// create client
+	bob, err := New(bobCtx, nil)
+	if err != nil {
+		fmt.Println("Bob client:", err)
+	}
+
+	// register for action events
+	bobActions := make(chan service.DIDCommAction)
+
+	err = bob.RegisterActionEvent(bobActions)
+	if err != nil {
+		fmt.Println("Bob RegisterActionEvent:", err)
+	}
+
+	// Handle actions
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		for {
+			select {
+			case <-aliceDone:
+				return
+			case action := <-aliceActions:
+				var (
+					thID string
+					// nolint: govet
+					err error
+				)
+
+				thID, err = action.Message.ThreadID()
+				if err != nil {
+					fmt.Println("Message.ThreadID", err)
+				}
+
+				action.Continue(alice.InvitationEnvelope(thID))
+				fmt.Println("alice received", action.Message.Type())
+			}
+		}
+	}()
+
+	// Handle actions
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		for {
+			select {
+			case <-bobDone:
+				return
+			case action := <-bobActions:
+				var (
+					thID string
+					// nolint: govet
+					err error
+				)
+
+				thID, err = action.Message.ThreadID()
+				if err != nil {
+					fmt.Println("Message.ThreadID", err)
+				}
+
+				fmt.Println("bob received", action.Message.Type())
+				action.Continue(alice.InvitationEnvelope(thID))
+			}
+		}
+	}()
+
+	// Send proposal
+	err = alice.SendProposalWithInvitation(&didexchange.Invitation{
+		Type: didexchange.InvitationMsgType},
+		&introduce.Recipient{
+			To: &introduce.To{
+				Name: "Bob",
```

Comment:
Should this be "Bob" ? This seems like we are sending request to "Bob" (`TheirDID` field) to introduce to "Bob".

Reply:
It is just a name, It can be whatever. I may change it to Carol.

Reply:
That would give more clarity.